services/oauth/src/main/java/org/mule/services/oauth/internal/builder/AbstractOAuthDancerBuilder.java

```diff
@@ -71,24 +67,69 @@ public OAuthDancerBuilder clientCredentials(String clientId, String clientSecret
   @Override
   public OAuthDancerBuilder tokenUrl(String tokenUrl) {
     this.tokenUrl = tokenUrl;
+    this.httpClientFactory = () -> {
+      final Builder clientConfigBuilder =
+          new HttpClientConfiguration.Builder().setThreadNamePrefix(format("oauthToken.requester[%s]", tokenUrl));
+      return httpService.getClientFactory().create(clientConfigBuilder.build());
+    };
     return this;
   }
 
   @Override
-  public OAuthDancerBuilder scopes(String scopes) {
-    this.scopes = scopes;
+  public OAuthDancerBuilder tokenUrl(HttpClient httpClient, String tokenUrl) {
+    this.httpClientFactory = () -> {
+      return new HttpClient() {
+
+        @Override
+        public void stop() {
```

Comment:
this seems completely unnecesary. Why don't you just have a factory/supplier which returns the provided instance? What's the wrapper for?